scripts/libgfapi-python/libgfapi-python.sh

```diff
@@ -4,10 +4,13 @@ source env
 # if anything fails, we'll abort
 set -e
 
-# install a Gluster server
-yum -y install centos-release-gluster yum-utils
+set -x
+
+# Strangely, userspace-rcu-devel comes from EPEL. We might as well install pip
+# from EPEL.
+yum -y install epel-release yum-utils
```

Comment:
There should not be a reason to use EPEL? centos-release-gluster enables a repository that provides us with the right version of userspace-rcu. It is possible (happened before) that userspace-rcu from EPEL was in a different version than what the glusterfs packages need.

Reply:
> There should not be a reason to use EPEL?

Agreed, assuming all required dependencies are present outside of EPEL.

> centos-release-gluster enables a repository that provides us with the right version of userspace-rcu

The `centos-release-gluster` contains released versions of Gluster, not nightly RPMs. The `userspace-rcu` provided by `centos-release-gluster` is compatible with the Gluster RPMs present in the `centos-release-gluster` repo, not necessarily the nightly RPMs. The right approach is to include `userspace-rcu` in the nightly repo ? That way, users of nightly repo don't have to install centos-release-gluster.

Here's an output from a centos7 machine:

```sh
# yum repolist
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.fibergrid.in
 * extras: mirrors.fibergrid.in
 * updates: mirrors.fibergrid.in
repo id                                                                             repo name                                                                                           status
base/7/x86_64                                                                       CentOS-7 - Base                                                                                     9,911
centos-gluster41/7/x86_64                                                           CentOS-7 - Gluster 4.1 (Long Term Maintanance)                                                        136
extras/7/x86_64                                                                     CentOS-7 - Extras                                                                                     432
gluster-nightly-master/7/x86_64                                                     Gluster Nightly builds (master branch)                                                                 20
updates/7/x86_64                                                                    CentOS-7 - Updates                                                                                  1,540
repolist: 12,039
```

```sh
# yum install glusterfs-server glusterfs-cli glusterfs-api
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.fibergrid.in
 * extras: mirrors.fibergrid.in
 * updates: mirrors.fibergrid.in
Resolving Dependencies
--> Running transaction check
---> Package glusterfs-api.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
--> Processing Dependency: glusterfs(x86-64) = 6dev-0.99.gitb5f75e3.el7 for package: glusterfs-api-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: glusterfs-client-xlators(x86-64) = 6dev-0.99.gitb5f75e3.el7 for package: glusterfs-api-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: libgfrpc.so.0()(64bit) for package: glusterfs-api-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: libgfxdr.so.0()(64bit) for package: glusterfs-api-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: libglusterfs.so.0()(64bit) for package: glusterfs-api-6dev-0.99.gitb5f75e3.el7.x86_64
---> Package glusterfs-cli.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
---> Package glusterfs-server.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
--> Processing Dependency: glusterfs-fuse(x86-64) = 6dev-0.99.gitb5f75e3.el7 for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: rpcbind for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: liburcu-bp.so.1()(64bit) for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: liburcu-cds.so.1()(64bit) for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
--> Running transaction check
---> Package glusterfs.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
---> Package glusterfs-client-xlators.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
---> Package glusterfs-fuse.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
--> Processing Dependency: attr for package: glusterfs-fuse-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: psmisc for package: glusterfs-fuse-6dev-0.99.gitb5f75e3.el7.x86_64
---> Package glusterfs-libs.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
---> Package glusterfs-server.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
--> Processing Dependency: liburcu-bp.so.1()(64bit) for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: liburcu-cds.so.1()(64bit) for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
---> Package rpcbind.x86_64 0:0.2.0-44.el7 will be installed
--> Processing Dependency: libtirpc >= 0.2.4-0.7 for package: rpcbind-0.2.0-44.el7.x86_64
--> Processing Dependency: libtirpc.so.1()(64bit) for package: rpcbind-0.2.0-44.el7.x86_64
--> Running transaction check
---> Package attr.x86_64 0:2.4.46-13.el7 will be installed
---> Package glusterfs-server.x86_64 0:6dev-0.99.gitb5f75e3.el7 will be installed
--> Processing Dependency: liburcu-bp.so.1()(64bit) for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
--> Processing Dependency: liburcu-cds.so.1()(64bit) for package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64
---> Package libtirpc.x86_64 0:0.2.4-0.10.el7 will be installed
---> Package psmisc.x86_64 0:22.20-15.el7 will be installed
--> Finished Dependency Resolution
Error: Package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64 (gluster-nightly-master)
           Requires: liburcu-bp.so.1()(64bit)
Error: Package: glusterfs-server-6dev-0.99.gitb5f75e3.el7.x86_64 (gluster-nightly-master)
           Requires: liburcu-cds.so.1()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

What do you suggest we do ?